### ISSUE TITLE
PHP 8 Support?

### ISSUE THREAD

**Issue by realdo-msk** [facebookarchive/php-graph-sdk#1217]
Hello.
Is there any plans for compatibility with php 8 in the near future?

**Comment by limonte** [comment]
This project is in https://github.com/facebookarchive

The latest commit was 2 years ago and nobody is answering issues. So, the most probable answer is "no".

**Comment by fd6130** [comment]
Oh my god.. any forks repo that still active and maintaining?

**Comment by holtkamp** [comment]
You might want to +1 https://github.com/facebookarchive/php-graph-sdk/pull/1212

**Comment by martinbean** [comment]
So if this is “archived”, what _is_ the official PHP SDK for Facebook?

**Comment by fd6130** [comment]
> So if this is “archived”, what _is_ the official PHP SDK for Facebook?

The facebook developer page lead us to here... so i think this is the official PHP SDK but  "archived".

**Comment by martinbean** [comment]
I swear this is like the _fourth_ time I’ve seen a PHP SDK created and archived. Surely engineers at Facebook have heard of versioning? The Graph API itself is versioned…

**Comment by JimmyBanks** [comment]
So does Facebook want us removing Facebook login systems for our PHP websites, or what is going on with this library being archived?

**Comment by JimmyBanks** [comment]
> I swear this is like the _fourth_ time I’ve seen a PHP SDK created and archived. Surely engineers at Facebook have heard of versioning? The Graph API itself is versioned…

Do you recall if they "archived" with no replacement prior to reintroducing a replacement?

**Comment by fd6130** [comment]
> So does Facebook want us removing Facebook login systems for our PHP websites, or what is going on with this library being archived?

We still can use this library for PHP websites but i think they won't maintain it anymore since it is "archived". Hope someone will fork this repo and continue the maintainance.

**Comment by hemberger** [comment]
Since there seems to not be any officially supported PHP SDK for Facebook, I've migrated to https://github.com/thephpleague/oauth2-facebook, which has been updated recently to include PHP 8 support. Works fine, and the migration was pretty painless.

**Comment by lesterchan** [comment]
Thanks @hemberger, I also migrated. Quite straightforward.

**Comment by martinbean** [comment]
> Hope someone will fork this repo and continue the maintainance.

@fd6130 I sincerely fail to believe that Facebook has the resources to maintain a PHP version of its API’s SDK. It’s not like PHP is a fringe or barely-used language on the web…

**Comment by bcastagna** [comment]
Any news on this issue ? I cant believe what I'm reading that FB just doesnt care anymore about this SDK ?!

**Comment by limonte** [comment]
I can't believe everybody is complaining about FB and nobody can fork the repo and add the PHP 8 support 😆

**Comment by vincentcastagna** [comment]
I can't believe Facebook still point to this SDK in their official documentation while its not maintained nor secure anymore.

**Comment by bcastagna** [comment]
@limonte I cannot see a single reason why you would say something this stupid. It's FACEBOOK SDK, not an opensource project, it's their responsibility to provide us developers with tools so we can access their products.

**Comment by limonte** [comment]
> @limonte I cannot see a single reason why you would say something this stupid. It's FACEBOOK SDK, not an opensource project, it's their responsibility to provide us developers with tools so we can access their products.

@bcastagna please avoid abusive language and read the project's licence: https://github.com/facebookarchive/php-graph-sdk/blob/5.x/LICENSE

I also recommend reading Facebook's Open Source Code of Conduct which you're violating with your language: https://engineering.fb.com/codeofconduct/

**Comment by bcastagna** [comment]
@limonte My "abusive" language was the fair response to your ironic notice, which was clearly passive-agressive and brought nothing constructive to solve the issue.
Yes, Licence-wise, perhaps, you are right, but it's just stupid when you own the biggest social niet-work of the world of not providing tools for developers to integrate your social network into their apps. It's stupid and counter-productive. Period.

**Comment by JimmyBanks** [comment]
> > @limonte I cannot see a single reason why you would say something this stupid. It's FACEBOOK SDK, not an opensource project, it's their responsibility to provide us developers with tools so we can access their products.
> 
> @bcastagna please avoid abusive language and read the project's licence: https://github.com/facebookarchive/php-graph-sdk/blob/5.x/LICENSE
> 
> I also recommend reading Facebook's Open Source Code of Conduct which you're violating with your language: https://engineering.fb.com/codeofconduct/

Your comments are intentionally trolling and not productive to the issue at hand, please refrain from continuing to troll.

**Comment by JimmyBanks** [comment]
@limonte Based on your first reply it should be you who volunteers to fork and maintain this repo.

If that isn't the case, then your comments are intentionally trolling and not productive to the issue at hand, please refrain from continuing.

**Comment by fd6130** [comment]
> I can't believe everybody is complaining about FB and nobody can fork the repo and add the PHP 8 support

Even though this is a sarcasm but i think this is too over for almost half of the users who really doesn't know how to add PHP 8 support to this sdk.

You're the one saying but even you didn't add PHP 8 support as well. Leave it be if you are not using this sdk, php or not going to help on this sdk.

**Comment by bobmagicii** [comment]
fwiw it works fine on php8 at least what little bit we used for facebook sign in. when you do composer anything just add --ignore-platform-reqs.

bcastagna is not wrong, its facebooks job to handle this other than the fact we all know they've bait and switched us now on the long game (suggested reading: history of facebook's use of php and development of hack lang). we just ended up removing facebook auth from our app and pressed on.

**Comment by spekulatius** [comment]
I tend to agree it's on FB to provide an SDK, if they want people to use their APIs. If this one will not be maintained anymore, at least provide a clearly described option/path to change. Guessing around isn't ideal.

**Comment by martinbean** [comment]
Any chance of an official response, @gfosco? You seem to be the contributor with the highest number of commits that’s still at Facebook.

**Comment by gfosco** [comment]
I don't work there any longer, sorry.  I kept it alive as long as I could.

**Comment by martinbean** [comment]
Thanks for replying, @gfosco. Do you have a contact who _could_ look at this? It seems a bit strange that Facebook don’t have the capacity to consistently maintain an SDK for perhaps the most popular web programming language on the web.

**Comment by spekulatius** [comment]
For those looking for an alternative: I've successfully implemented my required functionality using [facebook-php-business-sdk](https://github.com/facebook/facebook-php-business-sdk) and it's graph API. I can't promise it will work for you, but might be worth looking at.

**Comment by bobmagicii** [comment]
someone should do this as a joke at their f8 event and then walk out with the 100,000 for having fixed probably the previously most widely used social media library for php.

![image](https://user-images.githubusercontent.com/881944/116501530-9f784a00-a876-11eb-87a8-f3f0cde1404e.png)

**Comment by rajatvarlani** [comment]
You can check the software at the link https://wordpress.org/plugins/super-socializer, if you want to integrate Facebook login with your website. It's compatible with PHP 8 and it offers social login with 23 other social networks, like - Google, Linkedin, Twitter, Vkontakte, Line, Windows Live (Microsoft), Yahoo, WordPress, Dribbble, Kakao, Spotify, Github, Twitch, Instagram, Steam, Dropbox, Foursquare, Disqus, Reddit, Stack Overflow, Discord, Mail.ru and Amazon.

**Comment by martinbean** [comment]
> if you want to integrate Facebook login with your website.

@rajatvarlani We don’t. We want an SDK for Facebook’s entire Graph API that is supported for more than five minutes.

**Comment by stanislav-janu** [comment]
Hi guys, I created a refactored project for PHP 8.0+. Let's check https://github.com/janu-software/facebook-php-sdk 
You can contribute.
